src/avatar.jsx

```diff
@@ -156,7 +154,7 @@ const Avatar = React.createClass({
           className={className}
         >
           {icon && React.cloneElement(icon, {
-            color: color,
+            color: color || avatar.color,
```

Comment:
I agree but in this case there is. In the following line we are merging in the icons style with the style defined for the icon in this component. Which means if they provide a color there it should work.